3scale/porta
# 10579: Bot protection: Fix all auth strategies

**What this PR does / why we need it**:

https://github.com/3scale/porta/pull/3659 broke OAuth2 and SSO Token login strategies. This fixes the regression.

**Which issue(s) this PR fixes** 

[THREESCALE-10579](https://issues.redhat.com/browse/THREESCALE-10579)

**Notes**

AFAIK, there are 4 auth strategies for the Developer Portal

- `internal`: Provide username + password from the login screen
- `oauth2`: Using external authentication providers configured from the "Single Sign-on Integrations" form at `/p/admin/authentication_providers`
- `sso`: Using a generated auth token, as documented here: https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.14/html-single/creating_the_developer_portal/index#sso
- `cas`: Some old and probably unused auth method, I don't know about it.

Note how confusing is we internally call `oauth2` to something we call "Single Sign-on Integrations" in the UI; and at the same time we internally call `sso` to something which has nothing to do with the "SSO Integrations" screen or SSO at all.

The commit that actually fixes the bug is https://github.com/3scale/porta/pull/3669/commits/859aff4e223e93fa227581c87412beee4b9f174a. It disables the bot check for all auth strategies except `internal`.

I also refactored the cucumbers to ensure we have integration tests for `internal`, `oauth` and `sso`. In commits https://github.com/3scale/porta/pull/3669/commits/5c23b5d169e4ccfbf63d54d30af26e0a5cb82cd0 and https://github.com/3scale/porta/pull/3669/commits/40367f68984c6dec37ed5f4cb69a044b98a7c7f5

**Verification steps**

- Internal: Follow the instructions at https://github.com/3scale/porta/pull/3659. It should work the same way
- OAuth2: Configure any OAuth ID providers, for instance GitHub, as explained here: https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.14/html-single/creating_the_developer_portal/index#auth-github
  - When the process requires a new Sign Up, that should work as well.
- SSO Token: Foolow the instructions here: https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.14/html-single/creating_the_developer_portal/index#sso
